omnipaxos/src/storage.rs

```diff
@@ -693,23 +697,48 @@ where
         }
     }
 
+    fn create_decided_snapshot(&mut self) -> StorageResult<T::Snapshot> {
+        let log_decided_idx = match self.stopsign_is_decided() {
+            true => self.get_decided_idx() - 1,
+            false => self.get_decided_idx(),
+        };
+        self.create_snapshot(log_decided_idx)
+    }
+
     pub(crate) fn get_snapshot(&self) -> StorageResult<Option<T::Snapshot>> {
         self.storage.get_snapshot()
     }
 
+    // Creates a Delta snapshot of entries from `from_idx` to the end of the decided log and also
+    // returns the compacted idx of the created snapshot. If the range of entries contains entries
+    // which have already been compacted a valid delta cannot be created, so creates a Complete
+    // snapshot of the entire decided log instead.
     pub(crate) fn create_diff_snapshot(
```

Comment:
I'm a bit confused regarding this function. It's called create *diff* but we seem to return `Complete` snapshots?

Reply:
Imagine the follower has decided_idx = 2 and the leader decided_idx = 10 but also compacted_idx 5. The follower needs a snapshot representing entries 2-10, but the leader can't create that diff since it has compacted entries 1-5, so the best it can do is merge its own snapshot(1-5) with snap(6-10) = snap(1-10). Its a Complete snapshot because the follower needs to adopt snap(1-10) as is and not merge it onto its own snapshot like it would with a delta.

Reply:
Right, makes sense